src/mkdocstrings/handlers/base.py

```diff
@@ -386,6 +400,28 @@ def __init__(self, config: dict) -> None:
         """
         self._config = config
         self._handlers: Dict[str, BaseHandler] = {}
+        self._url_map = {}
+
+    def register_anchor(self, page: str, anchor: str):
+        """
+        Register that an anchor corresponding to an identifier was encountered when rendering the page.
+
+        Arguments:
+            page: The URL of the current page.
+            anchor: The HTML anchor (without '#') as a string
+        """
+        self._url_map[anchor] = f"{page}#{anchor}"
+
+    def __getitem__(self, identifier: str) -> str:
+        try:
+            return self._url_map[identifier]
+        except KeyError:
+            for handler in self._handlers.values():
+                try:
+                    return self._url_map[handler.renderer.get_anchor(handler.collector.collect(identifier, {}))]
```

Comment:
What happens if two objects from two files in different languages have the same identifier?

Reply:
As written here, quite simply, whichever handler was ever mentioned first gets priority. The same problem could exist even in the previous state (equal canonical identifiers + whichever identifier got mentioned last), but now a clash becomes a little more likely, but only in the case where neither language's canonical identifier was matched, but both language's "lax" lookup matched.

It seems not so horrible, but can warrant discussion.

Reply:
Hmmm I guess this is a general issue within projects injecting documentation for two or more languages.

Reply:
The cross-reference features expects all identifiers and their aliases to be unique. Maybe it's just not powerful enough yet.

Reply:
Yup. Maybe we could support a handler prefix in the identifier. And use the default handler only when there's no prefix.